posthog/mixins.py

```diff
@@ -1,38 +0,0 @@
-from rest_framework import mixins
-
-from posthog.event_usage import report_user_action
-
-
-def log_deletion_metadata_to_posthog(func):
-    """
-    wraps a DRF destroy endpoint and sends a PostHog event recording its deletion
-    so:
-        * args[0] is the ViewSet
-        * args[1] is the HTTP request
-    """
-
-    def wrapper(*args, **kwargs):
-        instance = args[0].get_object()
-        user = args[1].user
-        metadata = instance.get_analytics_metadata() if hasattr(instance, "get_analytics_metadata") else {}
-
-        func_result = func(*args, **kwargs)
-
-        report_user_action(user, f"{instance._meta.verbose_name} deleted", metadata)
-
-        return func_result
-
-    return wrapper
-
-
-class AnalyticsDestroyModelMixin(mixins.DestroyModelMixin):
```

Comment:
This is a cool idea but only used in one place and it led to 30 mins tracking down the bug. Removing it as it clearly didn't take off as an idea.